src/js/components/validateData/ValidateDataFileComponent.jsx

```diff
@@ -222,6 +225,72 @@ export default class ValidateDataFileComponent extends React.Component {
         return icon;
     }
 
+    signReport(item){
+        let type = null;
+        switch(item.file_type){
+            case 'appropriations':
+                type='A';
+                break;
+            case 'program_activity':
+                type='B';
+                break;
+            case 'award_financial':
+                type='C';
+                break;
+            case 'detached_award':
+                type='D2_detached';
+                break;
+            default:
+                break;
+        }
+        if(type){
+            GenerateFilesHelper.fetchFile(type, this.props.submission.id)
+            .then((result)=>{
+                this.setState({
+                    signInProgress: false,
+                    signedUrl: result.url
+                }, () => {
+                    this.openReport();
+                });
+            })
+            .catch((err) => {
+                this.setState({
+                    signInProgress: false
+                });
+                console.log(err);
+            });    
+        }else{
+            console.log('Invalid File type selected: '+item.file_type)
```

Comment:
Just console logging if we don't have the right type isn't going to help anyone except a dev. There should be some indication to the user that something is wrong rather than the button just not working for some reason